packages/account/src/providers/resource-cache.ts

```diff
@@ -1,33 +1,67 @@
 import { ErrorCode, FuelError } from '@fuel-ts/errors';
 import { hexlify } from '@fuel-ts/utils';
 
+import { GlobalCache } from './global-cache';
 import type { ExcludeResourcesOption } from './resource';
 
-interface CachedResource {
+export const DEFAULT_RESOURCE_CACHE_TTL = 20_000; // 20 seconds
+export const DEFAULT_RESOURCE_CACHE_STRATEGY = 'global';
+
+export interface CachedResource {
   utxos: Set<string>;
   messages: Set<string>;
   timestamp: number;
 }
 
-const cache = new Map<string, CachedResource>();
+export type CacheStrategy = 'global' | 'instance';
 
+/**
+ * Resource cache
+ */
 export class ResourceCache {
-  readonly ttl: number;
-
-  constructor(ttl: number) {
-    this.ttl = ttl; // TTL in milliseconds
+  private readonly ttl: number;
+  private readonly strategy: CacheStrategy;
+  private instanceCache: Map<string, CachedResource> = new Map();
+  private globalCache: GlobalCache;
```

Comment:
Why does `ResourceCache` store an instance of `GlobalCache`, even though it might not be used when the `instance` strategy is selected?